libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,284 @@
+    local ans = {ansx, ansy ,ansz}
+    return ans 
+end 
+
+function Leg_IK(X , Y , Z)
+    local coxa = math.atan(X,Y)* 180/math.pi
+    local trueX = math.sqrt(X^2+ Y^2 ) - L_COXA
+    local im = math.sqrt(trueX^2 + Z^2)
+
+    local q1 = -math.atan(Z,trueX)
+    local d1 = L_FEMUR^2 - L_TIBIA^2 + im^2
+    local d2 = 2*L_FEMUR*im
+    local q2 = math.acos(d1/d2)
+    local femur = (q1+q2) * 180/math.pi
+
+    local d1 = L_FEMUR^2 - im^2 + L_TIBIA^2
+    local d2 = 2*L_TIBIA*L_FEMUR
+    local tibia = (math.acos(d1/d2)-1.57) * 180/math.pi
+    local ang = { coxa, -femur ,-tibia}
+    return ang 
+end
+    
+function servo_estimate(start_time,current,last_angle)
+    local target = 0
+    for j = 1, 12 do
+        curr_target = math.abs(current[j] - last_angle[j])
+        if curr_target > target then
+            target = curr_target
+        end
+    end    
+    local target_time = target * servo_speed * 1000 
+    local now = millis()
+
+    if (target_time + start_time) <= now then
+        return true
+    else
+        return false
+    end   
+end
+
+function main_IK()
+
+  
+    local ans1 = Body_FK(endpoints1[1]+GaitPosX[1], endpoints1[2]+GaitPosY[1], endpoints1[3]+GaitPosZ[1], L/2, W/2,GaitRotZ[1])
+    local angles1 = Leg_IK(endpoints1[1]+ans1[1]+GaitPosX[1],endpoints1[2]+ans1[2]+GaitPosY[1], endpoints1[3]+ans1[3]+GaitPosZ[1])
+    angles1 = {-45 + angles1[1],angles1[2],angles1[3]}
+
+    local ans2 = Body_FK(endpoints2[1]+GaitPosX[2], endpoints2[2]+GaitPosY[2], endpoints2[3]+GaitPosZ[2], L/2, -W/2,GaitRotZ[2])
+    local angles2 = Leg_IK(endpoints2[1]+ans2[1]+GaitPosX[2],endpoints2[2]+ans2[2]+GaitPosY[2], endpoints2[3]+ans2[3]+GaitPosZ[2])
+    angles2 = {-135 + angles2[1],angles2[2],angles2[3]}
+
+    local ans3 = Body_FK(endpoints3[1]+GaitPosX[3], endpoints3[2]+GaitPosY[3], endpoints3[3]+GaitPosZ[3], -L/2, -W/2,GaitRotZ[3])
+    local angles3 = Leg_IK(endpoints3[1]+ans3[1]+GaitPosX[3],endpoints3[2]+ans3[2]+GaitPosY[3], endpoints3[3]+ans3[3]+GaitPosZ[3])
+    angles3 = {135 + angles3[1],angles3[2],angles3[3]}
+
+    local ans4 = Body_FK(endpoints4[1]+GaitPosX[4], endpoints4[2]+GaitPosY[4], endpoints4[3]+GaitPosZ[4], -L/2, W/2,GaitRotZ[4])
+    local angles4 = Leg_IK(endpoints4[1]+ans4[1]+GaitPosX[4],endpoints4[2]+ans4[2]+GaitPosY[4], endpoints4[3]+ans4[3]+GaitPosZ[4])
+    angles4 = {45 + angles4[1],angles4[2],angles4[3]}
+    Gaitselect()
+    current = {angles1[1],angles1[2],angles1[3],angles2[1],angles2[2],angles2[3],angles3[1],angles3[2],angles3[3],angles4[1],angles4[2],angles4[3]}
+  
+    if servo_estimate(start_time,current,last_angle) then
+        start_time = millis()
+        Sequence_Gen()
+        last_angle = current
+    end
+
+    return angles1,angles4,angles3,angles2
+end
+
+local pwm = { 1500, 1500, 1500,
+              1500, 1500, 1500,
+              1500, 1500, 1500,
+              1500, 1500, 1500 }
+
+
+function update()
+
+    X_SPEED = vehicle:get_control_outputs(1) * 0.4
```

Comment:
we should be more consistent about case. So for example "X_SPEED" is all capitals while other variables use camel case (like GaitType).  In ArduPilot we generally use all small letters with underscore between words.  So it would be "x_speed" and "gait_type".